src/ng/compile.js

```diff
@@ -2386,8 +2412,20 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             removeControllerBindingWatches =
               initializeDirectiveBindings(controllerScope, attrs, controller.instance, bindings, controllerDirective);
           }
+
+          if (isObject(controllerDirective.require) && !isArray(controllerDirective.require)) {
+            var controllerObj = getControllers(name, controllerDirective.require, $element, elementControllers);
+            extend(controller.instance, controllerObj);
```

Comment:
I meant if one of the **required** controllers returns a value, then we might extend **this** controller with the wrong value (depends on the order of execution of the directives).

(Your test proves that it works if **this** controller returns a value.)